src/java.base/share/classes/sun/security/ssl/SSLLogger.java

```diff
@@ -118,43 +137,49 @@ private static void help() {
 
     /**
      * Return true if the "javax.net.debug" property contains the
-     * debug check points, or System.Logger is used.
+     * debug check points, "all" or if the System.Logger is used.
+     *
+     * Specify all string tokens required when calling this method.
+     * E.g. since "plaintext" is a widened option of the "record" option,
+     * the call needs to be isOn("ssl,record,plaintext") to ensure
+     * correct use. It also ensures that the user specifies the correct
+     * system property value syntax as per help menu.
      */
     public static boolean isOn(String checkPoints) {
-        if (property == null) {              // debugging is turned off
+
+        if (!isOn) {
             return false;
-        } else if (property.isEmpty()) {     // use System.Logger
+        }
+
+        if (property.isEmpty() || property.equals("all")) {
+            // System.Logger in use or property = "all"
+            return true;
+        }
+
+        if (checkPoints.equals("ssl")) {
```

Comment:
No, the `ssl `option is designed to return all debug statements except those that are extra verbose (`data`, `packet`, `plaintext`). At least that's how the old (JDK 7) and newer TLSv1.3 stacks have been implemented. (with the exception of bug 8044609 which is being addressed here!)

The `record`, `handshake`, `keygen`, `session`, `defaultctx`, `sslctx`, `keymanager`, `trustmanager ` options have always struck me as odd. I reckon no one uses them. The only logic I can see with them is as filtering operations, since the `all `or `ssl `option is mandatory.

I think the filtering option (if we choose to keep them) is the only purpose we can have for these sub options. 

I can't imagine telling users to use `ssl` for some TLS debugging but turn on `record` or `handshake` or `keygen` or `session` or `defaultctx` or `sslctx` or `keymanager` or `trustmanager` options if you fancy other bits of TLS debugging.

Happy to address your CSR comments shortly. Let's agree on the implementation some more first. Nice suggestion on that EMPTYALL value. I'll get that change in shortly.

Reply:
> I can't imagine telling users...if you fancy other bits of TLS debugging.

That's exactly what I thought it was previously.  Have a look at the usage examples in the JSSE Ref Guide, [...To view the hexadecimal dumps of each handshake message, and to print trust manager tracing...](https://docs.oracle.com/en/java/javase/23/security/java-secure-socket-extension-jsse-reference-guide.html#GUID-31B7E142-B874-46E9-8DD0-4E18EC0EB2CF)

This is just like what is done in the JDK `java.security.debug` property in [Troubleshooting Security](https://docs.oracle.com/en/java/javase/23/security/troubleshooting-security.html#GUID-4F9F7BB5-5C9E-4888-9E28-3017EF746FA0)